frontend/simple/views/components/Contribution/Contribution.vue

```diff
@@ -0,0 +1,171 @@
+    <i18n tag="button" class="button has-text-primary" v-if="isAdding && isFilled" @click="handleSubmit">Add</i18n>
+    <i18n tag="button" class="button has-text-primary" v-if="isEditing && isFilled" @click="handleSubmit">Save</i18n>
+    <i18n tag="button" class="button has-text-danger" v-if="isEditing && !isFilled" @click="handleSubmit">Delete</i18n>
+  </li>
+
+  <li v-else-if="isEditable" :class="itemClasses">
+    <p class="box-body">
+      <slot></slot>
+    </p>
+    <div class="box-controls">
+      <button
+        class="button is-icon"
+        :aria-label="editAriaLabel"
+        @click="handleEditClick"
+      >
+        <i class="fa" :class="iconClass" aria-hidden="true"></i>
+      </button>
+    </div>
+  </li>
+
+  <li v-else-if="isUnfilled">
+    <button :class="itemClasses" @click="handleClick">
+      <slot></slot>
+    </button>
+  </li>
+
+  <li v-else :class="itemClasses">
+    <slot></slot>
+  </li>
+</template>
+<style lang="scss" scoped>
+@import "../../../assets/sass/theme/index";
+
+.c-contribution,
+.c-form {
+  min-height: 45px;
+}
+
+.c-contribution {
+  width: 100%;
+  font-size: inherit;
+
+  &.is-default {
+    border: none;
+    background-color: $white-ter;
+  }
+
+  &.is-editable {
+    border: none;
+    background-color: $primary-bg-a;
+  }
+
+  &.is-unfilled {
+    cursor: pointer;
+    text-align: left;
+
+    &:hover,
+    &:focus {
+      background-color: $primary-bg-a;
+    }
+  }
+
+  &-icon {
+    color: $primary;
+    margin-right: $gi-spacer-sm;
+  }
+}
+</style>
+<script>
+export default {
+  name: 'Contribution',
+  props: {
+    variant: {
+      type: String,
+      validator (value) {
+        return [
+          'default', // grey box just to read the content
+          'unfilled', // dashed box interactive to add a new contribution
+          'editable' // blue box to edit a contribution
+        ].indexOf(value) > -1
+      },
+      default: 'default'
+    }
+  },
+  data () {
+    return {
+      isAdding: false,
+      isEditing: false,
+      isFilled: null, // when true, show add/save button.
+      placeholders: [this.L('Portuguese classes'), this.L('Programming'), this.L('Cooking'), this.L('Parties'), this.L('Free cinema tickets')]
+    }
+  },
+  computed: {
+    itemClasses () {
+      return [
+        'box is-compact c-contribution',
+        `is-${this.variant}`,
+        { 'has-controls': this.isEditable }
+      ]
+    },
+    isEditable () {
+      return this.variant === 'editable'
+    },
+    isUnfilled () {
+      return this.variant === 'unfilled'
+    },
+    editAriaLabel () {
+      return this.L('Edit contribution settings')
+    },
+    randomPlaceholder () {
+      return this.placeholders[Math.floor(Math.random() * this.placeholders.length)]
+    },
+    iconClass () {
+      return this.$listeners.click ? 'fa-ellipsis-v' : 'fa-edit'
+    }
+  },
+  methods: {
+    handleClick () {
+      if (this.$listeners.click) {
```

Comment:
Note: I don't like very much this condition, but it was the cleaner way I've found to conditionally trigger a different behaviour when the contribution is monetary.
Maybe a prop `isMonetary` is more intuitive...

Reply:
I'm not sure about this `this.$listeners` approach... it seems very confusing and arbitrary to me.

With this approach, there is no clear relationship between the parent component and this one. It's not clear semantically what it means for `this.$listeners.click` to exist, so I would not rely on it for semantic meaning as is being done here.

Instead, I would recommend using `$emit` to emit events that the parent component can listen for and act appropriately.

Reply:
Okay, I'll use the `$emit` like I did with `@new-value`, actually it's way obvious.

Reply:
But nevertheless it's needed to confirm if the parent is listening to the $emit. When it's not, it should set `this.isAdding` to true (that shows the input)